### DISCUSSION TITLE
How to change the version of the currently existing lambda function?

### DISCUSSION THREAD

**Discussion by anne-montoya** [aws/copilot-cli#3920] [Q&A] [answered]
I understand that this https://github.com/aws/copilot-cli/issues/3896 was released on v1.21.0
I have an existing service and I redeployed it using `copilot svc deploy` but it didn't change my lambda files (or it's node version)

Should I change the node version of my lambda files manually or is there an existing command which can help me to do that?

Thanks

**Comment by dannyrandall** [comment, marked as answer]
Hey @anne-montoya! The change you mentioned only updates the node version for the lambdas that Copilot uses for some of it's functionality. If you are deploying your own lambdas through addons, you'll need to manually update the [`Runtime`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-runtime) field and redeploy (`copilot svc deploy`) your service.

--- 
Since you're using lambda and v1.21, you should check out [another new feature](https://aws.github.io/copilot-cli/docs/developing/addons/package/) in that release! It should help simplify the process to deploy a lambda using Copilot.

**Reply by xinye2005** [reply]
Hi @dannyrandall, so we still have to manually update the runtime field of the **existing** Lambda functions generated by copilot like EnvControllerFunction, RulePriorityFunction, DynamicDesiredCountFunction, DNSDelegationFunction & CertificateValidationFunction? And #3897 will only take effect for the new services created using copilot?

**Reply by dannyrandall** [reply]
Sorry @xinye2005, maybe I misunderstood @anne-montoya's question. If you have an existing service deployed with Copilot, it's lambda functions can be updated by doing the following:
1. Make sure `copilot version` is v1.21.0 (or higher)
2. Run `copilot env deploy` for each environment the service is deployed in. If you don't already have an environment manifest, follow [this guide](https://aws.github.io/copilot-cli/blogs/release-v120/#migrating-existing-environments) to generate one.
3. Run `copilot svc deploy`, again for each environment the service is deployed in

**Reply by xinye2005** [reply]
Many thanks @dannyrandall!

**Reply by anne-montoya** [reply]
Thanks @dannyrandall!
